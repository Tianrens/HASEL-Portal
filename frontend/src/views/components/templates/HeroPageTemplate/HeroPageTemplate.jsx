import React from 'react';
import { fullPage, rightSide } from './HeroPageTemplate.module.scss';
import GradientBg from '../../background/GradientBg/GradientBg';
import BlurredImageBg from '../../background/BlurredImageBg/BlurredImageBg';

function HeroPageTemplate({ children }) {
    return (
        <div className={fullPage}>
            <GradientBg>{children}</GradientBg>

            <div className={rightSide}>
                <BlurredImageBg />
            </div>
        </div>
    );
}

export default HeroPageTemplate;
