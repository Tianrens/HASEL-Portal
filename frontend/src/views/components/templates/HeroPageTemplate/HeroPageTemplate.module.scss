.fullPage {
    display: flex;
    width: 100vw;
    min-height: 100vh;
    height: 100%;
}

.leftSide {
    background: linear-gradient(to bottom, #fff 0%, rgba(0, 0, 0, 0) 100%);
    display: flex;
    flex-direction: column;
}

.rightSide {
    width: 35vw;
}
