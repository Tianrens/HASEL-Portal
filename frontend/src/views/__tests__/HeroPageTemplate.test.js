import React from 'react';
import renderer from 'react-test-renderer';
import { StylesProvider } from '@material-ui/core';
import HeroPageTemplate from '../components/templates/HeroPageTemplate/HeroPageTemplate';

test('HeroPageTemplate renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <HeroPageTemplate />
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
