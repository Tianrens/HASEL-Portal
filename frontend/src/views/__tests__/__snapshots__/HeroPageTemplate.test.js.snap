// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroPageTemplate renders properly 1`] = `
<div
  className="fullPage"
>
  <div
    className="heroPattern"
  >
    <div
      className="container"
    />
  </div>
  <div
    className="rightSide"
  >
    <div
      className="bgImg"
    />
  </div>
</div>
`;
